### PR TITLE
test: use gracePeriodSeconds=1 in e2e test while deleting pod

### PR DESCRIPTION
**Pull request by litaocdl**
closes: #2050

### CODE REVIEW

File: tests/e2e/asserts_test.go [hunk 1]
```diff
@@ -2281,10 +2281,16 @@ func OfflineResizePVC(namespace, clusterName string, timeout int) {
...
-				AssertClusterIsReady(namespace, clusterName, timeout, env)
 			}
 		}
+		// Make sure the cluster phase changed first
+		AssertClusterEventuallyDepartsPhase(namespace, clusterName, apiv1.PhaseHealthy, timeout)
+		// Ensuring cluster is healthy with three pods, we only need verify here once to safe time,
+		// but we need to make sure the cluster changed to un-health first, otherwise, it will be too
+		// quick, the status of the cluster did not have time to change from healthy to un-health, but
+		// the code reach to deleting primary pod.
+		AssertClusterIsReady(namespace, clusterName, timeout, env)
```
Comment (jsilvela) [inline tests/e2e/asserts_test.go:2292]: checking for cluster ready after cluster is no longer healthy (after pods have been deleted) is surprising me

File: tests/e2e/asserts_test.go [hunk 2]
```diff
@@ -2281,10 +2281,16 @@ func OfflineResizePVC(namespace, clusterName string, timeout int) {
 				// Deleting standby and replica pods
 				err = env.DeletePod(namespace, pod.Name, forceDelete)
 				Expect(err).ToNot(HaveOccurred())
-				// Ensuring cluster is healthy with three pods
-				AssertClusterIsReady(namespace, clusterName, timeout, env)
 			}
 		}
+		// Make sure the cluster phase changed first
+		AssertClusterEventuallyDepartsPhase(namespace, clusterName, apiv1.PhaseHealthy, timeout)
```
Comment (jsilvela) [inline tests/e2e/asserts_test.go:2287]: are we sure this is always going to happen? I'm testing locally and the transition isn't there...

**Review by jsilvela — APPROVED** [review]

File: tests/e2e/asserts_test.go (hunk 1 above)
Comment (mnencia) [reply tests/e2e/asserts_test.go:2314]: I think we should do it differently. AssertClusterIsReady should make sure that none of the Cluster's pods is deleted.

File: tests/e2e/asserts_test.go (hunk 2 above)
Comment (litaocdl) [reply tests/e2e/asserts_test.go]: Yeah, if we change grace_period=1, this test case failed everytime.

**Review by mnencia — APPROVED** [review]

**Pull request merged**